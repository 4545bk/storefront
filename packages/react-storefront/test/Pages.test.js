/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import Pages from '../src/Pages'
import { Provider } from 'mobx-react'
import AppModelBase from '../src/model/AppModelBase'
import { createMemoryHistory } from 'history'
import TestProvider from './TestProvider'

describe('Pages', () => {

  it('displays the correct page', () => {
    const component = (
      <TestProvider app={{ page: 'Category' }}>
        <Pages
          components={{
            Product: () => <div>Product</div>,
            Category: () => <div>Category</div>
          }}
        />
      </TestProvider>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('can display no page at all', () => {
    const component = (
      <TestProvider app={{ page: 'Foo' }}>
        <Pages
          components={{
            Product: () => <div>Product</div>,
            Category: () => <div>Category</div>
          }}
        />
      </TestProvider>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('displays a custom default loading component', () => {
    const component = (
      <TestProvider app={{ page: 'Product', loading: true }}>
        <Pages
          loading={() => <div>Loading...</div>}
          components={{
            Product: () => <div>Product</div>,
            Category: () => <div>Category</div>
          }}
        />
      </TestProvider>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('displays the default loading component', () => {
    const component = (
      <TestProvider app={{ page: 'Product', loading: true }}>
        <Pages
          components={{
            Product: () => <div>Product</div>
          }}
        />
      </TestProvider>
    )
    expect(mount(component)).toMatchSnapshot()
  })

  it('displays a custom skeleton', () => {
    const component = (
      <TestProvider app={{ page: 'Product', loading: true }}>
        <Pages
          loadMasks={{
            Product: () => <div id="product_load_mask">Loading Product...</div>
          }}
          components={{
            Product: () => <div>Product</div>
          }}
        />
      </TestProvider>
    )

    expect(mount(component).find('#product_load_mask').length).toBe(1)
  })

  it('caches the previous page in the dom', () => {
    const Product = () => <div id="product">Product</div>
    const Category = () => <div id="category">Category</div>
    const Loading = () => <div>Loading...</div>
    const app = AppModelBase.create({
      location: {
        pathname: '/',
        search: '',
        port: '80',
        hostname: 'localhost'
      },
      page: 'Product'
    })

    const wrapper = mount(
      <Provider app={app} history={createMemoryHistory()}>
        <Pages loading={Loading} components={{ Product, Category }} />
      </Provider>
    )

    app.applyState({ page: 'Category' })
    wrapper.update()
    expect(wrapper.find('#product').length).toBe(1)
    expect(wrapper.find('#category').length).toBe(1)
  })
})