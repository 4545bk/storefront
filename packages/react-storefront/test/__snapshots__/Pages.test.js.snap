// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages can display no page at all 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProvider theme={{...}}>
      <WithStyles(inject-Pages) components={{...}}>
        <inject-Pages components={{...}} classes={{...}}>
          <Pages components={{...}} classes={{...}} app={{...}} page="Foo" loading={false} uri="/" router={{...}} history={{...}} loadMasks={{...}}>
            <div className="RSFPages-root-16">
              <WithStyles(LoadMask) show={false} fullscreen={true}>
                <LoadMask show={false} fullscreen={true} classes={{...}}>
                  <div style={[undefined]} className="RSFLoadMask-root-18 RSFLoadMask-fullscreen-19">
                    <WithStyles(CircularProgress) className={[undefined]} color="secondary">
                      <CircularProgress className={[undefined]} color="secondary" classes={{...}} disableShrink={false} size={40} thickness={3.6} value={0} variant="indeterminate">
                        <div className="MuiCircularProgress-root-21 MuiCircularProgress-colorSecondary-25 MuiCircularProgress-indeterminate-23" style={{...}} role="progressbar">
                          <svg className="MuiCircularProgress-svg-26" viewBox="22 22 44 44">
                            <circle className="MuiCircularProgress-circle-27 MuiCircularProgress-circleIndeterminate-29" style={{...}} cx={44} cy={44} r={20.2} fill="none" strokeWidth={3.6} />
                          </svg>
                        </div>
                      </CircularProgress>
                    </WithStyles(CircularProgress)>
                  </div>
                </LoadMask>
              </WithStyles(LoadMask)>
            </div>
          </Pages>
        </inject-Pages>
      </WithStyles(inject-Pages)>
    </MuiThemeProvider>
  </Provider>
</TestProvider>
`;

exports[`Pages displays a custom default loading component 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProvider theme={{...}}>
      <WithStyles(inject-Pages) loading={[Function: loading]} components={{...}}>
        <inject-Pages loading={[Function: loading]} components={{...}} classes={{...}}>
          <Pages loading={true} components={{...}} classes={{...}} app={{...}} page="Product" uri="/" router={{...}} history={{...}} loadMasks={{...}}>
            <div className="RSFPages-root-31">
              <WithStyles(LoadMask) show={true} fullscreen={true}>
                <LoadMask show={true} fullscreen={true} classes={{...}}>
                  <div style={[undefined]} className="RSFLoadMask-root-33 RSFLoadMask-show-35 RSFLoadMask-fullscreen-34">
                    <WithStyles(CircularProgress) className={[undefined]} color="secondary">
                      <CircularProgress className={[undefined]} color="secondary" classes={{...}} disableShrink={false} size={40} thickness={3.6} value={0} variant="indeterminate">
                        <div className="MuiCircularProgress-root-36 MuiCircularProgress-colorSecondary-40 MuiCircularProgress-indeterminate-38" style={{...}} role="progressbar">
                          <svg className="MuiCircularProgress-svg-41" viewBox="22 22 44 44">
                            <circle className="MuiCircularProgress-circle-42 MuiCircularProgress-circleIndeterminate-44" style={{...}} cx={44} cy={44} r={20.2} fill="none" strokeWidth={3.6} />
                          </svg>
                        </div>
                      </CircularProgress>
                    </WithStyles(CircularProgress)>
                  </div>
                </LoadMask>
              </WithStyles(LoadMask)>
              <div data-page="Product" style={{...}}>
                <Product onBefore={[Function]} onAfter={[Function]}>
                  <div>
                    Product
                  </div>
                </Product>
              </div>
            </div>
          </Pages>
        </inject-Pages>
      </WithStyles(inject-Pages)>
    </MuiThemeProvider>
  </Provider>
</TestProvider>
`;

exports[`Pages displays the correct page 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProvider theme={{...}}>
      <WithStyles(inject-Pages) components={{...}}>
        <inject-Pages components={{...}} classes={{...}}>
          <Pages components={{...}} classes={{...}} app={{...}} page="Category" loading={false} uri="/" router={{...}} history={{...}} loadMasks={{...}}>
            <div className="RSFPages-root-1">
              <WithStyles(LoadMask) show={false} fullscreen={true}>
                <LoadMask show={false} fullscreen={true} classes={{...}}>
                  <div style={[undefined]} className="RSFLoadMask-root-3 RSFLoadMask-fullscreen-4">
                    <WithStyles(CircularProgress) className={[undefined]} color="secondary">
                      <CircularProgress className={[undefined]} color="secondary" classes={{...}} disableShrink={false} size={40} thickness={3.6} value={0} variant="indeterminate">
                        <div className="MuiCircularProgress-root-6 MuiCircularProgress-colorSecondary-10 MuiCircularProgress-indeterminate-8" style={{...}} role="progressbar">
                          <svg className="MuiCircularProgress-svg-11" viewBox="22 22 44 44">
                            <circle className="MuiCircularProgress-circle-12 MuiCircularProgress-circleIndeterminate-14" style={{...}} cx={44} cy={44} r={20.2} fill="none" strokeWidth={3.6} />
                          </svg>
                        </div>
                      </CircularProgress>
                    </WithStyles(CircularProgress)>
                  </div>
                </LoadMask>
              </WithStyles(LoadMask)>
              <div data-page="Category" style={{...}}>
                <Category onBefore={[Function]} onAfter={[Function]}>
                  <div>
                    Category
                  </div>
                </Category>
              </div>
            </div>
          </Pages>
        </inject-Pages>
      </WithStyles(inject-Pages)>
    </MuiThemeProvider>
  </Provider>
</TestProvider>
`;

exports[`Pages displays the default loading component 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProvider theme={{...}}>
      <WithStyles(inject-Pages) components={{...}}>
        <inject-Pages components={{...}} classes={{...}}>
          <Pages components={{...}} classes={{...}} app={{...}} page="Product" loading={true} uri="/" router={{...}} history={{...}} loadMasks={{...}}>
            <div className="RSFPages-root-46">
              <WithStyles(LoadMask) show={true} fullscreen={true}>
                <LoadMask show={true} fullscreen={true} classes={{...}}>
                  <div style={[undefined]} className="RSFLoadMask-root-48 RSFLoadMask-show-50 RSFLoadMask-fullscreen-49">
                    <WithStyles(CircularProgress) className={[undefined]} color="secondary">
                      <CircularProgress className={[undefined]} color="secondary" classes={{...}} disableShrink={false} size={40} thickness={3.6} value={0} variant="indeterminate">
                        <div className="MuiCircularProgress-root-51 MuiCircularProgress-colorSecondary-55 MuiCircularProgress-indeterminate-53" style={{...}} role="progressbar">
                          <svg className="MuiCircularProgress-svg-56" viewBox="22 22 44 44">
                            <circle className="MuiCircularProgress-circle-57 MuiCircularProgress-circleIndeterminate-59" style={{...}} cx={44} cy={44} r={20.2} fill="none" strokeWidth={3.6} />
                          </svg>
                        </div>
                      </CircularProgress>
                    </WithStyles(CircularProgress)>
                  </div>
                </LoadMask>
              </WithStyles(LoadMask)>
              <div data-page="Product" style={{...}}>
                <Product onBefore={[Function]} onAfter={[Function]}>
                  <div>
                    Product
                  </div>
                </Product>
              </div>
            </div>
          </Pages>
        </inject-Pages>
      </WithStyles(inject-Pages)>
    </MuiThemeProvider>
  </Provider>
</TestProvider>
`;
